src/router.ts

```diff
@@ -0,0 +1,44 @@
+import { $, $$ } from './utils';
+
+interface IRouter {
+  path: string;
+  component: Element;
+}
+
+const nav = document.querySelector('.nav');
+const baseURL = '/javascript-vendingmachine';
+
+nav.addEventListener('click', (e) => {
+  historyRouterPush((e.target as HTMLElement).getAttribute('route'));
+});
+
+const historyRouterPush = (pathname: string) => {
+  history.pushState({ pathname }, '', pathname);
+  render(pathname);
+};
+
+const render = (path: string) => {
+  $$('.focus-button').forEach((button) => button.classList.remove('focus-button'));
+  $(`[route='${path}']`, nav)?.classList.add('focus-button');
+
+  const cur = routers.find((route) => route.path === path)?.component ?? $('product-management');
+  const prevs = routers.filter((route) => route.path !== path);
```

Comment:
```suggestion
  const currentComponent = routers.find((route) => route.path === path)?.component ?? $('product-management');
  const prevRoute = routers.filter((route) => route.path !== path);
```
변수명은 정확하게 적어주는게 좋을거같아요!

Reply:
넵~ 변수명을 정확하게 적어줘야 어떤 역할울 허눈 변수인지 바로 파악할 수 있을 것 같습니다!

Reply:
반영 링크입니다! [082edc3](https://github.com/woowacourse/javascript-vendingmachine/pull/32/commits/082edc31363bd66f8a6cbf67f9708f3fd981d97d)